libraries/provider_database_mysql_user.rb

```diff
@@ -84,6 +84,11 @@ def action_grant
         def exists?
           db.query("SELECT User,host from mysql.user WHERE User = '#{@new_resource.username}' AND host = '#{@new_resource.host}'").num_rows != 0
         end
+
+        def mysql_escape (s)
+          # having in mind that escaped string will be used within single quotes!
+          s.gsub(/'/,"''").gsub(/\\/,'\\\\\\')
```

Comment:
I think it's better to use [`Mysql.escape_string`](http://www.rubydoc.info/gems/mysql/2.9.1/Mysql#escape_string-class_method) method here.